client/src/main/scala/com.gu.contentapi.client/model/PaginatedApiResponse.scala

```diff
@@ -19,23 +17,19 @@ trait PaginatedApiResponse[Response] {
 object PaginatedApiResponse {
 
   implicit val searchResponse = new PaginatedApiResponse[SearchResponse] {
-    def getNextId = r => if (r.pages == r.currentPage) None else r.results.lastOption.map(_.id)
-    def getPrevId = r => if (1 == r.currentPage) None else r.results.headOption.map(_.id)
+    def getNextId = r => if (r.results.length < r.pageSize) None else r.results.lastOption.map(_.id)
```

Comment:
`r.results.length < r.pageSize` is unexpected! Is it intentional? If so, could you please document it?

Reply:
I don't think it is. You can set `page-size` to be 100 and only get 50 results back.

Reply:
I see. It would be useful to see a comment that describes how this is the terminal condition for pagination: a page smaller than requested.

Nonetheless, we should beware the edge case where `r.total % r.pageSize == 0`, in which case this condition will fail.